src/python/pants/option/config_test.py

```diff
@@ -210,3 +308,46 @@ def test_empty(self) -> None:
     assert config.sources() == []
     assert config.has_section("DEFAULT") is False
     assert config.has_option(section="DEFAULT", option="name") is False
+
+
+class ConfigIniTest(ConfigBaseTest):
+  __test__ = True
+
+  file1_suffix = ".ini"
+  file2_suffix = ".ini"
+  file1_content = FILE1_INI
+  file2_content = FILE2_INI
+
+
+class ConfigTomlTest(ConfigBaseTest):
+  __test__ = True
+
+  file1_suffix = ".toml"
+  file2_suffix = ".toml"
+  file1_content = FILE1_TOML
+  file2_content = FILE2_TOML
+
+  @property
+  def default_file1_values(self):
+    return {**super().default_file1_values, "disclaimer": "Let it be known\nthat."}
+
+  @property
+  def expected_file1_options(self):
+    return {
+      **super().expected_file1_options,
+      "a": {
+        **super().expected_file1_options["a"], "list": '["1", "2", "3", "42"]',
+      },
+      "b.nested": {
+        "dict": '{\n  "a": 1,\n  "b": "42",\n  "c": ["42", "42"],\n}'
+      },
+    }
+
+
+class ConfigMixedTest(ConfigBaseTest):
+  __test__ = True
+
+  file1_suffix = ".ini"
+  file2_suffix = ".toml"
+  file1_content = FILE1_INI
+  file2_content = FILE2_TOML
```

Comment:
To be really comprehensive, we could add the reverse case (file1 is toml, file2 is ini). I'm only half kidding, but I only even thought of this because you've made it so easy to add new cases!! Thanks a lot for that.